paasta_tools/cli/cmds/logs.py

```diff
@@ -559,6 +610,86 @@ def cluster_to_scribe_env(self, cluster):
             return env
 
 
+def generate_start_end_time(from_string="30m", to_string=None):
+    """
+    Parses the --from and --to arguments to create python datetime objects representing
+    the start and end times for log retrieval
+    """
+    if to_string is None:
+        end_time = datetime.datetime.utcnow()
+    else:
+        # Try parsing as a a natural time duration first, if that fails move on to
+        # parsing as an ISO-8601 timestamp
+        to_duration = timeparse(to_string)
+
+        if to_duration is not None:
+            end_time = datetime.datetime.utcnow() - datetime.timedelta(seconds=to_duration)
+        else:
+            end_time = isodate.parse_datetime(to_string)
+
+    from_duration = timeparse(from_string)
+    if from_duration is not None:
+        start_time = datetime.datetime.utcnow() - datetime.timedelta(seconds=from_duration)
+    else:
+        start_time = isodate.parse_datetime(from_string)
+
+    # Covert the timestamps to something timezone aware
+    start_time = pytz.utc.localize(start_time)
+    end_time = pytz.utc.localize(end_time)
+
+    if start_time > end_time:
+        raise ValueError("Start time bigger than end time")
+
+    return start_time, end_time
+
+
+def validate_filtering_args(args, log_reader):
+    if not log_reader.SUPPORTS_LINE_OFFSET and args.line_offset is not None:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support line based offsets")
+        return False
+    if not log_reader.SUPPORTS_LINE_COUNT and args.line_count is not None:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support line count based log retrieval")
+        return False
+    if not log_reader.SUPPORTS_TAILING and args.tail:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support tailing")
+        return False
+    if not log_reader.SUPPORTS_TIME and (args.time_from is not None or args.time_to is not None):
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support time based offsets")
+        return False
+
+    if args.tail and (args.line_count is not None or args.time_from is not None or
+                      args.time_to is not None or args.line_offset is not None):
+        print PaastaColors.red("You cannot specify line/time based filtering parameters when tailing")
```

Comment:
Why not?

Both the `tail` command and Kafka consumers are capable of starting from N lines/messages since the end of the stream/file, printing those, then continuing on to follow.

Scribereader is the outlier.